artifacts/definitions/Windows/Remediation/Sinkhole.yaml

```diff
@@ -0,0 +1,187 @@
+name: Windows.Remediation.Sinkhole
+description: |
+   **Apply a Sinkhole via Windows hosts file modification**  
+   This content will apply modifications to the Windows hosts file by a 
+   configurable lookup table.  
+   During application, the configuration is backed up and used as the base for 
+   subsequent changes.  
+   
+   Parameters:  
+   HostsFile - path to hosts file
+   HostsFileBackup - name to backup original hosts file. If reapplying policy.  
+   this is the configuration used as base.  
+   CommentPrefix -  prefix to add to description in hosts file comments.  
+   RestoreBackup - checkbox to enable restoration of backup hosts file.  
+   SinkholeTable - table of Domains to add to or modify in hosts file.
+   
+   NOTE:
+   Modifying the hosts file may cause network communication issues. I have 
+   disabled any sinkhole settings on the Velociraptor configuration but there 
+   are no rail guards on other domains. Use with caution.
+
+author: Matt Green - @mgreen27
+
+required_permissions:
+  - EXECVE
+  
+type: CLIENT
+
+parameters:
+  - name: HostsFile
+    default: C:\Windows\System32\drivers\etc\hosts
+  - name: HostsFileBackup
+    default: C:\Windows\System32\drivers\etc\hosts.velociraptor.backup
+  - name: CommentPrefix
+    default: "Velociraptor sinkhole"
+  - name: RestoreBackup
+    description: "Restore hosts file backup"
```

Comment:
If this is ticked does it ignore all the other parameters and restore it anyway?

Should this be a different artifact? One to set and one to restore?

Reply:
see above. I am happy for a single artefact but happy to switch it up if you prefer a less complex workflow?